senders/client_factory.go

```diff
@@ -138,16 +147,19 @@ func createConfig(wfURL string, setters ...Option) (*configuration, error) {
 	return cfg, nil
 }
 
-// newWavefrontClient creates a Wavefront sender
-func newWavefrontClient(cfg *configuration) (Sender, error) {
+func newSender(cfg *configuration) (Sender, error) {
 	client := internal.NewClient(cfg.Timeout, cfg.TLSConfig)
 	metricsReporter := internal.NewReporter(cfg.metricsURL(), cfg.Token, client)
 	tracesReporter := internal.NewReporter(cfg.tracesURL(), cfg.Token, client)
 	sender := &wavefrontSender{
 		defaultSource: internal.GetHostname("wavefront_direct_sender"),
 		proxy:         !cfg.Direct(),
 	}
```

Comment:
Description: If internal metrics are enabled , create a `RealMetricsRegistry`. Otherwise create a `NoOpRegistry`.

Reply:
Now that we have `NewSender` calling `newSender`, I wounder if we can unline `newSender` into `NewSender`

Reply:
Can you explain what "unline" means? I tried keyword searching on the web, but I couldn't find a relevant definition.

Reply:
Oh no! That's a typo. I meant "inline", as in eliminating the unexported `newSender` method and putting that code directly in `NewSender`

Reply:
Cool, that makes sense.  I'd forgotten about the word "inline" -- thanks for the reminder!

Because `newSender` has so many lines in it, I find it pleasant to look at them separately:
- I think of the exported `NewSender` as creating the `configuration` struct. 
- I think of `newSender` as consuming values on the configuration struct to create reporters and line handlers. 

However I also think it would be pretty easy to break it apart again if ever needed, so I think I'll go with the inlining.

Reply:
I agree, `newSender` is a little long. I don't like having two functions whose names are only different by one character, and  I don't like that whatever makes `NewSender` and `newSender` different, the function names don't shed any light on it. If we can make that situation better, I'd be happy to have the two different functions.

Reply:
I hadn't thought of these arguments and I agree with them, too. Thanks for explaining. I inlined. 👍